app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt

```diff
@@ -0,0 +1,271 @@
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.material.icons.filled.Menu
+import androidx.compose.material.icons.filled.Warning
+import androidx.compose.material3.BottomSheetScaffold
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.rememberBottomSheetScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.viewmodel.compose.viewModel
+import ch.hikemate.app.R
+import ch.hikemate.app.model.route.ListOfHikeRoutesViewModel
+import org.osmdroid.config.Configuration
+import org.osmdroid.events.MapListener
+import org.osmdroid.events.ScrollEvent
+import org.osmdroid.events.ZoomEvent
+import org.osmdroid.util.GeoPoint
+import org.osmdroid.views.MapView
+
+object MapScreen {
+  const val TEST_TAG_MAP = "map"
+  const val TEST_TAG_MENU_BUTTON = "menuButton"
+  const val TEST_TAG_HIKES_LIST = "hikesList"
+  const val TEST_TAG_HIKE_ITEM = "hikeItem"
+}
+
+@Composable
+fun MapScreen(
+    hikingRoutesViewModel: ListOfHikeRoutesViewModel =
+        viewModel(factory = ListOfHikeRoutesViewModel.Factory)
+) {
+  val context = LocalContext.current
+  // Avoid re-creating the MapView on every recomposition
+  val mapView = remember { MapView(context) }
+
+  Configuration.getInstance().apply {
+    // Set user-agent to avoid rejected requests
+    userAgentValue = context.packageName
+
+    // Allow for faster loading of tiles. Default OSMDroid value is 2.
+    tileDownloadThreads = 4
+
+    // Maximum number of tiles that can be downloaded at once. Default is 40.
+    tileDownloadMaxQueueSize = 40
+
+    // Maximum number of bytes that can be used by the tile file system cache. Default is 600MB.
+    tileFileSystemCacheMaxBytes = 600L * 1024L * 1024L
+  }
+
+  mapView.apply {
+    controller.setZoom(15.0)
+    // TODO : Set the center to the user's location
+    controller.setCenter(GeoPoint(46.5, 6.6))
+    // Enable touch-controls such as pinch to zoom
+    setMultiTouchControls(true)
+    // Update hiking routes every time the user moves the map
+    // TODO : Those updates could be quite frequent, have a cooldown to avoid sending a request each
```

Comment:
Note: You can remove this todo, there is a task in the sprint backlog to fix this. Please however add a marker to the code and add it to the task in the sprint backlog, so the person working on the task knows where to look

Reply:
Actually this TODO is irrelevant based on what we discussed about #41. Since there will be a _Search Here_ button, those events do not need to be throttled anymore, because they won't be used.

I'll remove the TODO altogether. I added information about those event listeners in #41 directly (even though I'm assigned to it).